src/mon/OSDMonitor.cc

```diff
@@ -3106,7 +3106,7 @@ bool OSDMonitor::preprocess_command(MonOpRequestRef op)
     int64_t epochnum;
     cmd_getval(g_ceph_context, cmdmap, "epoch", epochnum, (int64_t)0);
     epoch = epochnum;
-    if (!epoch)
+    if (epoch > 0)
```

Comment:
this change breaks the behaviour of 
`ceph osd getmap`.

in this case, `epoch` is 0.